micro_sam/evaluation/benchmark_datasets.py

```diff
@@ -0,0 +1,499 @@
+        if not isinstance(dataset_choice, list):
+            dataset_choice = [dataset_choice]
+
+    for choice in dataset_choice:
+        if choice in available_datasets:
+            available_datasets[choice]()
+        else:
+            raise ValueError(f"'{choice}' is not a supported choice of dataset.")
+
+    return dataset_choice
+
+
+def _extract_slices_from_dataset(path, dataset_choice, crops_per_input=10):
+    if dataset_choice in [*LM_2D_DATASETS, *EM_2D_DATASETS]:
+        ndim, tile_shape = 2, (512, 512)
+    else:
+        ndim, tile_shape = 3, (32, 512, 512)
+
+    available_datasets = {
+        # Light Microscopy datasets
+        "livecell": lambda: datasets.livecell.get_livecell_paths(path=path, split="test"),
+        "deepbacs": lambda: datasets.deepbacs.get_deepbacs_paths(path=path, split="test", bac_type="mixed"),
+        "plantseg_root": lambda: datasets.plantseg.get_plantseg_paths(path=path, name="root", split="test"),
+        "plantseg_ovules": lambda: datasets.plantseg.get_plantseg_paths(path=path, name="ovules", split="test"),
+
+        # Electron Microscopy datasets
+        "lucchi": lambda: datasets.lucchi.get_lucchi_paths(path=path, split="test"),
+    }
+
+    if ndim == 2:
+        image_paths, gt_paths = available_datasets[dataset_choice]()
+
+        if dataset_choice in DATASET_RETURNS_FOLDER:
+            image_paths = glob(os.path.join(image_paths, DATASET_RETURNS_FOLDER[dataset_choice]))
+            gt_paths = glob(os.path.join(gt_paths, DATASET_RETURNS_FOLDER[dataset_choice]))
+
+        image_paths, gt_paths = natsorted(image_paths), natsorted(gt_paths)
+        assert len(image_paths) == len(gt_paths)
+
+        paths_set = zip(image_paths, gt_paths)
+
+    else:
+        image_paths = available_datasets[dataset_choice]()
+        if isinstance(image_paths, str):
+            paths_set = [image_paths]
+        else:
+            paths_set = natsorted(image_paths)
+
+    # Directory where we store the extracted ROIs.
+    save_image_dir = os.path.join(path, f"roi_{ndim}d", "inputs")
+    save_gt_dir = os.path.join(path, f"roi_{ndim}d", "labels")
+    if os.path.exists(save_image_dir) and os.path.exists(save_gt_dir):
+        return save_image_dir, save_gt_dir, ndim
+
+    os.makedirs(save_image_dir, exist_ok=True)
+    os.makedirs(save_gt_dir, exist_ok=True)
+
+    # Logic to extract relevant patches for inference
+    image_counter = 1
+    for per_paths in tqdm(paths_set, total=len(paths_set), desc=f"Extracting patches for {dataset_choice}"):
+        if ndim == 2:
+            image_path, gt_path = per_paths
+            image, gt = util.load_image_data(image_path), util.load_image_data(gt_path)
+        else:
+            image_path = per_paths
+            image = util.load_image_data(image_path, DATASET_CONTAINER_KEYS[dataset_choice][0])
+            gt = util.load_image_data(image_path, DATASET_CONTAINER_KEYS[dataset_choice][1])
+
+        skip_smaller_shape = (image.shape > tile_shape)
```

Comment:
Does this work? We need to do some elementwise comparison here, but `image.shape` and `tile_shape` are tuples. I think you need to case them to arrays and then use `.all`:
```python
skip_smaller_shape = (np.array(image.shape) >= np.array(tile_shape)).all()
```

Reply:
Oh yeah you are right. There's the case where the first dimension for image shape might be higher while the later matching, then this would fail. Your suggestion would work for all dimensions. Thanks for the mention!